### ISSUE TITLE
`error_on_unformatted` is ignored when set from rustfmt.toml

### ISSUE THREAD

**Issue by euclio** [rust-lang-nursery/rustfmt#2565]
```
$ rustfmt --version
rustfmt 0.4.1-nightly (374dba83 2018-03-22)
```

With the following source:

main.rs:
```
// Lorem ipsum dolor sit amet, consectetur adipiscing elit. Duis eget ex commodo, posuere sapien non, interdum leo.
fn main() {
    println!("Hello, world!");
}
```

rustfmt.toml:
```
error_on_unformatted = true
```

Notice that `rustfmt main.rs` (erroneously) succeeds, but `rustfmt main.rs --error-on-unformatted` fails (as expected).

**Comment by topecongiro** [comment]
https://github.com/rust-lang-nursery/rustfmt/blob/3de184fb0e6f7161fef2f1d9288e06f57abadfc0/src/bin/main.rs#L115-L127

`CliOptions::apply_to` erroneously overrides the config option values even when the corresponding command line arguments are not passed. We should use `Option<bool>` instead of `bool` for every `CliOptions` field, and only change config when we need to, like we already do for `write_mode` and `color`.

**Comment by i1skn** [comment]
I would like to work on this one if nobody has already started.

**Comment by i1skn** [comment]
I see that `from_matches` has some logic to define `unstable_features`, so it makes sense to use at as a bool in `apply_to`, right?

https://github.com/rust-lang-nursery/rustfmt/blob/3de184fb0e6f7161fef2f1d9288e06f57abadfc0/src/bin/main.rs#L74-L84

**Comment by nrc** [comment]
> I would like to work on this one if nobody has already started.

Great! And sorry for the delay on reply, I've been travelling.

> I see that from_matches has some logic to define unstable_features, so it should not be overridden from a config file, right?

Yes

> Also in Operation::Format I see verbose used before calling apply_to.
So, it should not be overridden from a config file too, right?

I think this could go either way. It might be that like now it should only be set on the command line, in which case we should make sure it is not documented as a config option and throws an error when used there. Alternatively, we could change the code here so that it can properly be used as a config option.

> As outcome only skip_children and error_on_unformatted should have Option<bool> type, is this make sense?

In the config or elsewhere? I think they can still be `bool` rather than `Option<bool>` (the command line should always override the config file, so we don't need to know whether we got the value from the default or an explicit value).

**Comment by topecongiro** [comment]
Closed via #2600.
